.scripts/summarize_progress/main.py

```diff
@@ -0,0 +1,138 @@
+    else:
+        lines_all = lines_tranlated + lines_untranlated
+        progress = lines_tranlated / lines_all
+        progress_percentage = round(progress * 100, 2)
+        result = f"Ongoing, {str(progress_percentage)} %"
+
+    return result
+
+
+def get_open_issues_count() -> int:
+    url = f"https://api.github.com/search/issues?q=repo:python/python-docs-zh-tw+type:issue+state:open"
+    headers = {
+        "Accept": "application/vnd.github+json",
+        "X-GitHub-Api-Version": "2022-11-28",
+    }
+    r = requests.get(url=url, headers=headers)
+    result = r.json()
+
+    return result["total_count"]
+
+
+def get_github_issues() -> list:
+    NUMBER_OF_ISSUES = get_open_issues_count()
+
+    url = f"https://api.github.com/search/issues?q=repo:python/python-docs-zh-tw+type:issue+state:open&per_page={NUMBER_OF_ISSUES}"
+    headers = {
+        "Accept": "application/vnd.github+json",
+        "X-GitHub-Api-Version": "2022-11-28",
+    }
+    r = requests.get(url=url, headers=headers)
+    result = r.json()
+
+    result_list = []
+    for issue in result["items"]:
+        title_segments = issue["title"].split()
+
+        if len(title_segments) < 2:
+            continue
+        if title_segments[0] != "翻譯" and title_segments[0].lower() != "translate":
+            continue
+        if issue["assignee"] is None:
+            continue
+
+        filename = title_segments[1].strip("`")
+
+        if re.fullmatch("[a-zA-z-]+/[a-zA-Z0-9._-]+.po", filename):
+            filename_segments = filename.split("/")
+            # print(filename_segments)
+        elif re.fullmatch("[a-zA-z-]+/[a-zA-Z0-9._-]+", filename):
+            filename_segments = filename.split("/")
+            filename_segments[1] += ".po"
+        else:
+            continue
+
+        result_list.append([filename_segments, issue["assignee"]["login"]])
+
+    return result_list
+
+
+def format_line_file(filename, result) -> str:
+    return f"  - {filename.ljust(37, '-')}{result}\r\n"
+
+
+def format_line_directory(dirname) -> str:
+    tmp = f"- {dirname}/\r"
+    return tmp
+
+
+if __name__ == "__main__":
+    issue_list = get_github_issues()
+
+    directories = [
+        "c-api",
+        "distributing",
+        "extending",
+        "faq",
+        "howto",
+        "includes",
+        "installing",
+        "library",
+        "reference",
+        "tutorial",
+        "using",
+        "whatsnew",
+    ]
+
+    summary = {}
+
+    file_list = glob.glob("./../**/*.po", recursive=True)
+    file_list.sort()
+
+    for filepath in file_list:
+        if len(filepath.split("/")) == 4:  # in-dir files
+            _, _, dirname, filename = filepath.split("/")
+        else:  # root dir files
+            _, _, filename = filepath.split("/")
+            dirname = "/"
+
+        if dirname not in summary:
+            summary[dirname] = {}
+
+        po = polib.pofile(filepath)
+        result = entry_check(po)
+        summary[dirname][filename] = result
```

Comment:
改寫供參考（但沒測試過XD）
- 希望不要有 hardcoded list of directory，因為有新增 directory 就要來改 script
- 如果檔案結構改變的話，用 slash 數量來判斷位置會使得 script 需要改比較多地方
- 可以善用 pathlib 的功能
```suggestion
    # from pathlib import Path
    BASE_DIR = Path("../")
    summary = {}
    for filepath in glob.glob(str(BASE_DIR / "**/*.po"), recursive=True):
        path = Path(filepath)
        filename = path.name
        dirname = path.parent.name if path.parent.name != BASE_DIR.name else '/'

        po = polib.pofile(filepath)
        summary.setdefault(dirname, {})[filename] = entry_check(po)
```

Reply:
測試過了沒問題，學到了好多筆記@@

Reply:
following with the suggestion above, it would be like this without sorting.

![image](https://github.com/python/python-docs-zh-tw/assets/34214497/556574e7-faad-442e-a14e-a4c0ca5c09a0)